tests/test_modeling_tf_common.py

```diff
@@ -995,10 +1011,10 @@ def test_lm_head_model_random_beam_search_generate(self):
             model = model_class(config)
 
             if config.bos_token_id is None:
-                # if bos token id is not defined mobel needs input_ids, num_return_sequences = 1
+                # if bos token id is not defined model needs input_ids, num_return_sequences = 1
                 self._check_generated_ids(model.generate(input_ids, do_sample=True, num_beams=2))
-            else:
-                # num_return_sequences = 1
+            elif model_class.__name__ not in ["TFSpeech2TextForConditionalGeneration"]:
+                # Models with non-text inputs won't work here; num_return_sequences = 1
```

Comment:
For those two tests, maybe we should overwrite the corresponding tests in the speech2text test file.